plugins/single_plugins/scale.cpp

```diff
@@ -652,16 +664,23 @@ class wayfire_scale : public wf::plugin_interface_t
                 {
                     vg = child->get_wm_geometry();
 
-                    scale_x = width / vg.width;
-                    scale_y = height / vg.height;
+                    double child_scale_x = width / vg.width;
+                    double child_scale_y = height / vg.height;
+                    child_scale_x = child_scale_y = std::min(
+                        std::min(scale_x, scale_y), max_scale_factor);
+                    if (max_scale_child > 0.0 && child_scale_x > max_scale_child * scale_x)
+                    {
+                        child_scale_x = max_scale_child * scale_x;
+                        child_scale_y = child_scale_y;
+                    }
```

Comment:
`child_scale_y = child_scale_y;` is definitely wrong. If we do limit scaling of children, we should just limit the max to 1.0. I see no reason for this added complexity.

However, I don't think we should limit children from scaling larger. Consider the case where the parent is scaled smaller but the child is smaller than the parent. I would think it would make the most sense to allow the child to become as big as possible so the user has the best chance of seeing it well. This is scale after all, not only-scale-down.

Reply:
Yes, this is a typo, I'm adding a fix.

Reply:
I wonder how I did not see this as an issue before

Reply:
It's probably because you never hit a case where it mattered.

The more I think about it, the more I think we should just calculate parent scale, clamp it to 1.0, and use the same scale for the child. Would this work?

Reply:
I think we need to handle the case when the child is bigger than the parent. In that case, the child must have a smaller scale than parent, otherwise it would not fit in the space allocated for them, so that's why I'm calculating a scale for the child first and then limiting it based on the parent.

E.g. with gedit, in the unlikely case that the main window is resized so small that the "Close without saving" dialog is larger.

Reply:
Good point. Carry on then :+1: